src/lib/features/snackbar/specs/snackbar.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { render } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { SnackbarQueue, snackbar, type SnackbarOptions } from '..';
+
+const firstSnackbar: SnackbarOptions = {
+  text: 'Hello',
+  buttonText: 'Close',
+  callback: jest.fn(),
+};
+
+const secondSnackbar: SnackbarOptions = {
+  text: 'Example',
+  buttonText: 'Close',
+  callback: jest.fn(),
+};
+
+it('adds a snackbar to the queue and displays it', () => {
+  snackbar(firstSnackbar);
+  const { getByText } = render(SnackbarQueue);
+  expect(getByText(firstSnackbar.text)).toBeInTheDocument();
```

Comment:
issue: usually, things that rely on time are particularly fussy to test, the tests for them tend to be flaky. Jest has a way to remedy that by letting you control time. See [this page](https://jestjs.io/docs/timer-mocks) for reference.

suggestion: add time control to avoid test flake

Reply:
Good idea, implemented this 👍

Reply:
But why did you only do it for one test? It's best if all tests will be time-independent